cmd/snap-mgmt/snap-mgmt.sh.in

```diff
@@ -239,6 +242,188 @@ purge() {
     fi
 }
 
+ensure_snap_apps_stopped() {
+    apps_or_services="$(find /sys/fs/cgroup/ -name 'snap.*.*.scope' -o -name 'snap.*.*.service')"
+    if [ -n "$apps_or_services" ]; then
+        echo "Found active snap services or applications in the following cgroups:"
+        for n in $apps_or_services; do
+            echo "- $n: PIDs: $(cat "$n/cgroup.procs" | paste -s -d' ')"
+        done
+        return 1
+    fi
+    return 0
+}
+
+discard_mount_namespaces() {
+    if [ ! -d /run/snapd/ns ]; then
+        return
+    fi
+
+    tooldir="$1"
+    if [ -z "$tooldir" ]; then
+        tooldir="$STATIC_SNAP_TOOLING_DIR"
+    fi
+
+    echo "Discarding snap mount namespaces"
+    find /run/snapd/ns/ -name '*.mnt' | while read -r mntns; do
+        snname="$(basename "${mntns%.mnt}")"
+        echo "  ..discarding mount namespace of snap $snname"
+        "$tooldir"/snap-discard-ns "$snname"
+    done
+}
+
+_mount_dir_migrate() {
+    mount_from="$1"
```

Comment:
show a disclaimer pointing to the forum as a place where issues should be reported